sky/skylet/providers/aws/config_v2.py

```diff
@@ -0,0 +1,1234 @@
+import copy
```

Comment:
yes, I have the same comment, we should use a feature flag. otherwise version control/compatibility in the future would be horrible.

once there is huge enough difference between us/Ray autoscaler, then we should write something to migrate old setting to new settings

Reply:
The main reason that I add a different config file is that I think using flag will involve more modifications in our forked  version. 
Previously, the workflow I used for updating our fork with the upper stream was to download the upper stream files we are currently based on (for example, ray v2.0.1) and do a `diff` between the two files, and adopt the changes to the new version of the file. I think adding a feature flag will generate more diff between the files, and making the process harder.
Also, it will be quite messy to use a flag, as we have to modify multiple global variables, making the code hard to read.

Now, `config.py` is actually deprecated and only used for backward compatibility. If we want to update the file from the upstream, we can do something similar to `config_v2.py` (no need to change `config.py`).

> otherwise version control/compatibility in the future would be horrible.
@suquark Could you explain how that will cause the problem and how could keep it in the same file help?

Reply:
I can try to add the flag to the original `config.py` file and let you guys compare the two methods if you guys think that will be easier to see the difference and decide which one to use. : )

Reply:
Thank you for your suggestions @suquark @concretevitamin! I just tried to merge the new changes to the original `config.py` and it looks ok without much modifications.

Tested:
1. on master branch `sky launch -c old echo hi`
2. check the IAM role of that instance to be `ray-autoscaler-v1` 
3. switch to the current branch `sky launch -c old echo hi`, the job ID is 2 means it reuses the original cluster (also checked the console that there is only 1 cluster named old)